src/consensus/aft/raft.h

```diff
@@ -505,59 +508,132 @@ namespace aft
+              data, size);
+          aee =
+            std::make_unique<AppendEntryResponseExecEntry>(this, std::move(r));
+        }
+        else if (type == raft_append_entries_signed_response)
+        {
+          SignedAppendEntriesResponse r =
+            channels->template recv_authenticated<SignedAppendEntriesResponse>(
+              data, size);
+          aee = std::make_unique<SignedAppendEntryResponseExecEntry>(
+            this, std::move(r));
+        }
 
-        case raft_request_vote:
-          recv_request_vote(data, size);
-          break;
+        else if (type == raft_request_vote)
+        {
+          RequestVote r =
+            channels->template recv_authenticated<RequestVote>(data, size);
+          aee = std::make_unique<RequestVoteExecEntry>(this, std::move(r));
+        }
 
-        case raft_request_vote_response:
-          recv_request_vote_response(data, size);
-          break;
+        else if (type == raft_request_vote_response)
+        {
+          RequestVoteResponse r =
+            channels->template recv_authenticated<RequestVoteResponse>(
+              data, size);
+          aee =
+            std::make_unique<RequestVoteResponseExecEntry>(this, std::move(r));
+        }
 
-        case bft_signature_received_ack:
-          recv_signature_received_ack(data, size);
-          break;
+        else if (type == bft_signature_received_ack)
+        {
+          SignaturesReceivedAck r =
+            channels->template recv_authenticated<SignaturesReceivedAck>(
+              data, size);
+          aee = std::make_unique<SignatureAckExecEntry>(this, std::move(r));
+        }
 
-        case bft_nonce_reveal:
-          recv_nonce_reveal(data, size);
-          break;
+        else if (type == bft_nonce_reveal)
+        {
+          NonceRevealMsg r =
+            channels->template recv_authenticated<NonceRevealMsg>(data, size);
+          aee = std::make_unique<NonceRevealExecEntry>(this, std::move(r));
+        }
+        else if (type == bft_view_change)
+        {
+          RequestViewChangeMsg r =
+            channels
+              ->template recv_authenticated_with_load<RequestViewChangeMsg>(
+                data, size);
+          aee = std::make_unique<ViewChangeExecEntry>(
+            this, std::move(r), data, size, std::move(d));
+        }
 
-        case bft_view_change:
-          recv_view_change(data, size);
-          break;
+        else if (type == bft_view_change_evidence)
+        {
+          ViewChangeEvidenceMsg r =
+            channels
+              ->template recv_authenticated_with_load<ViewChangeEvidenceMsg>(
+                data, size);
 
-        case bft_view_change_evidence:
-          recv_view_change_evidence(data, size);
-          break;
+          aee = std::make_unique<ViewChangeEvidenceExecEntry>(
+            this, std::move(r), data, size, std::move(d));
+        }
+        else
+        {
+          throw std::logic_error("Unknown message type");
```

Comment:
Also include `type` in error message here.

Reply:
In fact, this shouldn't be a `throw` but simply a `LOG_FAIL` as we do not want to crash a node if a message is corrupted (integrity verification only happens if the message type is valid).